### ISSUE TITLE
run-test-system.sh not producing expected output

### ISSUE THREAD

**Issue by jgarci40** [BinaryAnalysisPlatform/bap-plugins#6]
I am trying to get the run-test-system.sh script working in minos. However, it's not producing any output really.

I don't have IDA, so when I run the script in its original form, I get this:

```
minos$ ./run-test-system.sh 
Failed to extract symbols from IDA: Can't find IDA in your environment
Producing cut groups...

===================================
ROOTING at NTH 1 sink callstrings.
===================================
callstring sinks:

[+] 0 cut groups
Producing trims...

Processing paths...
```

However, if I remove the IDA option in the script, I still just get this:
```
minos$ ./run-test-system.sh 
Producing cut groups...

===================================
ROOTING at NTH 1 sink callstrings.
===================================
callstring sinks:

[+] 0 cut groups
Producing trims...

Processing paths...
```

Here are the files in the output directory:
```
minos$ ls -R analysis-test-system-1/
analysis-test-system-1/:
cut_groups  meta.txt  trim_groups

analysis-test-system-1/cut_groups:

analysis-test-system-1/trim_groups:
```

Here are the contents of meta.txt:
```
Src: LCA:1
Sink: @system

Mode: nth-from-sink

callstring sinks:

Cut groups: 0
```

Thank you for your time and consideration.

**Comment by ivg** [comment]
TL;DR; Remove `--no-byteweight` option.

Long story
========

In order to disassemble file, we first need to find functions. There're two ways to find them, first is to use IDA and rely on its excellent FLIRT algorithm, other is to use our own Byteweight algorithm, that is not bad also. Usually it is bad idea to use them both simultaneously, so @rvantonder has disabled Byteweight, and enabled IDA.  You have disabled the ida one, so you need to enable back the byteweight.

**Comment by rvantonder** [comment]
I'm afraid it's not that simple, since this plugin relies on symbol names. For instance, if generate IR output with IDA as intended (`bap --use-ida idaq64 --no-byteweight -dbir test1`), you'll see symbol names resolved:

```


`bap -dbir test1`

**Comment by ivg** [comment]
We actually should resolve such symbol names in BAP, this is just a PLT entry, no magic here and no FLIRT involved. Looks like that we became addicted to IDA.

**Comment by jgarci40** [comment]
Just to clarify, it sounds like minos currently requires IDA. Is that correct?

**Comment by rvantonder** [comment]
It is intended to be used with IDA until we resolve PLT symbols inside BAP. There is a temporary work-around, however. BAP accepts a file containing the symbol-address mapping. For example, here is the one that is generated for the example binary with the `bap --dump-symbols` option (you would need IDA to generate this):

```
(_IO_stdin_used 34420 34448)
(__libc_csu_fini 34408 34411)
(__libc_csu_init 34308 34347)
(__libc_csu_init 34396 34399)
(__libc_csu_init 34388 34395)
(__libc_csu_init 34364 34387)
(__libc_csu_init 34356 34363)
(__libc_csu_init 34348 34355)
(main 34064 34107)
(main 34280 34287)
(main 34276 34279)
(main 34252 34275)
(main 34240 34251)
(main 34208 34239)
(main 34176 34207)
(main 34172 34175)
(main 34140 34171)
(main 34108 34139)
(sub_8474 33908 33939)
(sub_8474 33956 33959)
(sub_8474 33952 33955)
(sub_8474 33940 33951)
(sub_8418 33816 33839)
(sub_8418 33840 33843)
(sub_8418 33720 33731)
(sub_83c4 33732 33743)
(system 33708 33719)
(sub_8394 33684 33695)
(sub_8374 33652 33659)
(sub_8374 33660 33663)
```

The good news is you can craft this file manually (or with a script), by using objdump if you are only interested in PLT entries. Then you feed this file to BAP with `--syms=my_sym_file`.

Resolve PLT entries with objdump with `objdump --syms binary/test-system/test1`. In the output you'll see 
```000083ac       F *UND*  00000000              system@@GLIBC_2.4```

So you can create a symbol file with contents `(system 0x83ac 0x83ac)`. The second hex value should actually be the end address of the function, but you may need to do extra work to figure that out. Anyway, by duplicating the start address, it will at least mimic the intended behavior. Altogether:

`symbols.syms`:
```
(system 0x83ac 0x83ac)
```

`run-test-system.sh`:
```
rm -rf analysis-test-system-*

bap --syms=symbols.syms -lminos.plugin binary/test-system/test1 \
  --minos-check=system \
  --minos-out_dir=analysis-test-system-1 \
  --minos-srcs=configs/n-1-@system/srcs.txt \
  --minos-sinks=configs/n-1-@system/sinks.txt \
  --minos-with_dots
```

The example should now work for you.
